src/rbd_fuse/rbd-fuse.cc

```diff
@@ -877,13 +865,59 @@ connect_to_cluster(rados_t *pcluster)
 	return r;
 }
 
+int init_cluster(int argc, char **argv, char **remargv)
+{
+	int r, remargc;
+
+	r = rados_create(&cluster, NULL);
+	if (r < 0) {
+		simple_err("Could not create cluster handle", r);
+		exit(1);
+	}
+
+	r = rados_conf_parse_argv_remainder(cluster, argc, (const char **)
+						argv, (const char **) remargv);
+	if (r < 0) {
+		simple_err("Error parsing the arguments", r);
+		exit(1);
+	}
+
+	for (remargc = 0; remargv[remargc] != NULL and remargc < argc;
+		++(remargc))
+		;
+
+	return remargc;
+}
+
+void post_init_cluster()
+{
+	int r;
+
+	r = rados_conf_read_file(cluster, rbd_options.ceph_config);
+	if (r < 0) {
+		simple_err("Error reading Ceph config file", r);
+		exit(1);
+	}
+
+	r = rados_conf_parse_env(cluster, NULL);
+	if (r < 0) {
+		simple_err("Error parsing the environment", r);
+		exit(1);
+	}
+}
+
 int main(int argc, char *argv[])
 {
-	struct fuse_args args = FUSE_ARGS_INIT(argc, argv);
+	char *remargv[argc] = {NULL};
+	int remargc = init_cluster(argc, argv, remargv);
```

Comment:
nit: I would still prefer my version. When a function returns something, I always suppose this is the most important thing, e.g. the result or status (error code) of its work. And if a function may fail I would like to see this just looking on how it is used, without need to read this function code.

Saving one line or two is not always good if it makes the code less readable. But taking that this is rather a small  code, it is not important here, so it is up to you what signature to use.